comictaggerlib/ctsettings/__init__.py

```diff
@@ -25,3 +34,37 @@
     "ct_ns",
     "group_for_plugin",
 ]
+
+
+class SettingsEncoder(json.JSONEncoder):
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, pathlib.Path):
+            return str(obj)
+
+        # Otherwise raise standard error
+        raise TypeError(f"Object of type {obj.__class__.__name__} is not JSON serializable")
```

Comment:
I don't believe this will work I believe you need use the parent `default` method otherwise only Pathlib.Path objects will get converted

Reply:
Looking at the output bool, int etc. all seem to have been converted. As I understand it, it will try the internal supported conversions and then use the custom class. `default` only raises a type error in the base. I will change the error raise as suggest here though: https://docs.python.org/3/library/json.html#json.JSONEncoder.default